lib/src/components/NxModal/NxModal.tsx

```diff
@@ -44,10 +45,31 @@ const NxModal: FunctionComponent<Props> = ({className, onClose, variant, ...attr
     return removeCloseHandlerListener;
   }, [onClose]);
 
+  useEffect(function() {
+    if (dialogRef.current) {
+      const el = dialogRef.current;
+
+      /*
+       * This will cause the document to become "blocked by the modal dialog"
+       * (https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog)
+       * meaning that only the modal and its contents are interactable/focusable.
+       *
+       * Note: not supported in safari, which is why the conditional is here. Once safari gets
+       * it together, we should be able to take advantage of the "top layer" functionality built into
+       * the browser around modals to simplify the NxModal styling around z-index handling
+       */
+       if (el.showModal) {
+        el.showModal();
+      }
+    }
+  }, []);
+
   return (
-    <div className="nx-modal-backdrop">
-      <section className={modalClasses} {...attrs} />
-    </div>
+    // Note: role="dialog" should be redundant but I think some screenreaders (ChromeVox) don't know
+    // what a <dialog> is.  It makes a difference there.
+    <dialog ref={dialogRef} role={role || 'dialog'} aria-modal="true" className="nx-modal-backdrop">
```

Comment:
This makes me really uncomfortable. Neither FF or Safari support `<dialog>`, and they way it's being used/styled here isn't as a working element but as the backdrop.